app/views/posts/show.html.erb

```diff
@@ -5,4 +5,45 @@
   <span>
     <%= @post.user.first_name %>
   </span>
-</article>
\ No newline at end of file
+
+</article>
+
+<% if @post.likes.find_by(user_id: current_user.id)%>
```

Comment:
- [ ] There should be no application logic in the views:

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png)

**`find_by`** method is business or (application logic) and the right place for it is to be encapsulated in its active record model
We need to move it to the model and use it in here.

- [ ] Also there should be no logic at all in the views
- [ ] Rails has built-in mechanism to deal with view-related logic which is View Helpers (app/helpers/*_helper.rb). So the common convention is to move any logic in the views to the helpers. Please take a look at [this](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) for more information.